### PR TITLE
#386 Changed rendering of parameters values to quoted in content-disposition header

### PR DESCRIPTION
**Pull request by KrzyMucha**
…on header according to rfc2616

### CODE REVIEW

**Comment by ktoso** [comment]
OK TO TEST

**Comment by ktoso** [comment]
Resolves #386

File: akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/headers.scala [hunk 1]
```diff
@@ -383,13 +383,18 @@ final case class `Content-Length` private[http] (length: Long) extends jm.header
   protected def companion = `Content-Length`
 }
 
-// http://tools.ietf.org/html/rfc6266
+/**
+ * Document http://tools.ietf.org/html/rfc6266 updates document https://www.w3.org/Protocols/rfc2616/rfc2616-sec19.html. Between this
```
Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/headers.scala]: `between these`

File: akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/headers.scala [hunk 2]
```diff
@@ -383,13 +383,18 @@ final case class `Content-Length` private[http] (length: Long) extends jm.header
...
+ * two there is slight but important difference regarding how parameter values are formatted. In rfc6266 this values are without quotes and
+ * in rfc2616 they are quoted. Since common practice among http servers is to understand quoted values, we use older document
```
Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/headers.scala]: `RFC2616`

File: akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/headers.scala (hunk 1 above)
Comment (KrzyMucha) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/headers.scala]: fixed

File: akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/headers.scala (hunk 2 above)
Comment (KrzyMucha) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/headers.scala]: fixed

**Comment by jrudolph** [comment]
Compilation fails with:

```
[error] /localhome/jenkinsakka/workspace/pr-validator-akka-http/akka-http-tests/src/test/scala/akka/http/scaladsl/unmarshalling/MultipartUnmarshallersSpec.scala:280: value withCharset is not a member of akka.http.scaladsl.model.MediaType.Multipart
[error]           `multipart/form-data` withBoundary "XYZABC" withCharset `UTF-8`,
[error]                                                       ^
[warn] four warnings found
```

I wonder why it works in the other files...

**Comment by jrudolph** [comment]
I think it's because this PR is based on a version from a few weeks ago. Can you rebase and fix the test, @KrzyMucha?

**Comment by KrzyMucha** [comment]
Yes, im doint it now

File: akka-http-tests/src/test/scala/akka/http/scaladsl/unmarshalling/MultipartUnmarshallersSpec.scala [hunk 3]
```diff
@@ -275,6 +275,17 @@ class MultipartUnmarshallersSpec extends FreeSpec with Matchers with BeforeAndAf
                        |--XYZABC--""".stripMarginWithNewline("\r\n")))).to[Multipart.FormData] should haveParts(
           Multipart.FormData.BodyPart.Strict("email", HttpEntity(`application/octet-stream`, ByteString("test@there.com"))))
       }
+      "with one element and name value in quotes" in {
+        Unmarshal(HttpEntity(
+          `multipart/form-data` withBoundary "XYZABC" withCharset `UTF-8`,
```
Comment (jrudolph) [inline akka-http-tests/src/test/scala/akka/http/scaladsl/unmarshalling/MultipartUnmarshallersSpec.scala]: Can you replace this with

```scala
(`multipart/form-data` withBoundary randomBoundary).toContentType
```

Comment (KrzyMucha) [inline akka-http-tests/src/test/scala/akka/http/scaladsl/unmarshalling/MultipartUnmarshallersSpec.scala]: I saw your comment after fixing. I made it little bit different but should be ok :)

Comment (jrudolph) [inline akka-http-tests/src/test/scala/akka/http/scaladsl/unmarshalling/MultipartUnmarshallersSpec.scala]: :+1:

**Comment by jrudolph** [comment]
Great contribution, thanks again @KrzyMucha.

**Pull request merged**